File: App2/src/app/_core/_model/oc.ts
export class Oc {
    constructor() {
        this.children = new Array<Oc>();
    }
    key: number;
    levelnumber: number;
    parentid: number;
    state: boolean;
    code: string;
    HasChildren: boolean;
    children: Array<Oc>;
}

export interface OC {
  id: number;
  no: string;
  type: string;
  name: string;
  principal: string;
  length: number;
  level: number;
  width: number;
  longtitude: string;
  latitude: string;
  temptureTopLimit: string;
  temptureLowLimit: string;
  photoPath: string;
  ggp: number;
  gp: number;
  pmpf: number;
  semen: number;
  nursery: number;
  grower: number;
  comment: string;
  parentID: number | null;
  status: boolean | null;
  createBy: number;
  updateBy: number | null;
  deleteBy: number | null;
  createDate: string;
  updateDate: string | null;
  deleteDate: string | null;
  file: any;
}
export class HierarchyNode<T> {
  childNodes: Array<HierarchyNode<T>>;
  depth: number;
  hasChildren: boolean;
  parent: T;
  constructor() {
      this.childNodes = new Array<HierarchyNode<T>>();
  }
  any(): boolean {
      return this.childNodes.length > 0;
  }
}

File: PigFarm/Models/OC.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace PigFarm.Models
{
    public partial class Oc
    {
        public Oc()
        {
            Accounts = new HashSet<Account>();
        }

        public int Id { get; set; }
        public string Type { get; set; }
        public string Name { get; set; }
        public string Principal { get; set; }
        public int Length { get; set; }
        public int Width { get; set; }
        public int Ggp { get; set; }
        public int Gp { get; set; }
        public int Pmpf { get; set; }
        public int Semen { get; set; }
        public int Nursery { get; set; }
        public int Grower { get; set; }
        public string Comment { get; set; }
        public int? ParentId { get; set; }
        public bool? Status { get; set; }
        public int CreateBy { get; set; }
        public int? UpdateBy { get; set; }
        public int? DeleteBy { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime? UpdateDate { get; set; }
        public DateTime? DeleteDate { get; set; }
        public string No { get; set; }
        public int Level { get; set; }

        public virtual ICollection<Account> Accounts { get; set; }
    }
}
